### PR TITLE
Deprecate Finch Json.

### PR DESCRIPTION
**Pull request by rpless**
Added deprecation warnings and noted the deprecation in documentation. A start for #233. (should be noted that this adds a lot of warnings to our build).

Do we want to include updates to the demo package in this PR or do we want to do that when we remove the library entirely?

### CODE REVIEW

File: docs/json.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### Finch-JSON
 
+This library is deprecated in favor of other json libraries.
```
Comment (travisbrown) [inline docs/json.md]: One bit of nitpicking—elsewhere "JSON" is capitalized in prose (also below).

**Comment by vkostyukov** [comment]
Yes. I think we should also update the demo package to use Argonaut JSON. @rpless do you mind to do that in this PR?

**Comment by travisbrown** [comment]
Looks good to me, but can you provide explicit arguments for all `@deprecation` annotations? Using named arguments for these annotations on 2.10 causes weird failures during test compilation (which you can see in the Travis CI results).

File: docs/json.md (hunk 1 above)
Comment (rpless) [inline docs/json.md]: Good catch. We should keep it consistent.

**Comment by rpless** [comment]
I'll make the updates to demo and fix the 2.10 compiler fails.

**Comment by rpless** [comment]
I've updated the documentation and the failing build. I'm still working on the updates. Its proving more than a little annoying because of package name collisions.

**Comment by rpless** [comment]
I've updated the demo. I tried to change as little as possible, so I used the argonaut json API instead of case class codecs. Thoughts?

File: demo/src/main/scala/io/finch/demo/model.scala [hunk 2]
```diff
@@ -35,12 +35,16 @@ object model {
 
   // A ticket object with two fields: `id` and `label`.
   case class Ticket(id: Long, label: String) extends ToJson {
-    override def toJson = Json.obj("id" -> id, "label" -> label)
+    override def toJson = Json.obj("id" -> jNumber(id), "label" -> jString(label))
```
Comment (travisbrown) [inline demo/src/main/scala/io/finch/demo/model.scala]: `Json("id" := id, "label" := label)` would be a better sell.

File: demo/src/main/scala/io/finch/demo/model.scala [hunk 3]
```diff
@@ -35,12 +35,16 @@ object model {
...
   }
 
   // A user object with three fields: `id`, `name` and a collection of `tickets`.
   case class User(id: Long, name: String, tickets: Seq[Ticket]) extends ToJson {
-    override def toJson = Json.obj("id" -> id, "name" -> name, "tickets" -> Json.arr(tickets.map(_.toJson) :_*))
+    override def toJson = Json.obj(
```
Comment (travisbrown) [inline demo/src/main/scala/io/finch/demo/model.scala]: Similarly here:

``` scala
    override def toJson = Json(
      "id" := id,
      "name" := name,
      "tickets" := tickets.map(_.toJson).toList
    )
```

Could also change the `Seq` to `List` and skip the `toList`.

File: demo/src/main/scala/io/finch/demo/model.scala [hunk 4]
```diff
@@ -57,7 +61,7 @@ object model {
    */
   implicit class SeqToJson[A](s: Service[A, Seq[ToJson]]) extends Service[A, ToJson] {
     private[demo] def seqToJson(seq: Seq[ToJson]) = new ToJson {
-      def toJson: Json = Json.arr(seq.map { _.toJson }: _*)
+      def toJson: Json = Json.array(seq.map { _.toJson }: _*)
```
Comment (travisbrown) [inline demo/src/main/scala/io/finch/demo/model.scala]: I'd probably use `jArray` here but that's more a matter of taste.

**Comment by travisbrown** [comment]
@rpless Thanks! This is looking good to me, but using the `Json("foo" := thingWithACodec)` syntax wherever possible would simplify it a bit further. I've noted a few places above, and would be happy to do a pass in a separate PR.

**Comment by rpless** [comment]
@travisbrown Thanks for the feedback. Making the updates now.

**Comment by travisbrown** [comment]
:+1: from me, thanks!

**Comment by vkostyukov** [comment]
This looks great! Thanks, Ryan, Travis!

**Pull request merged**